src/Core/CustomObjects/CustomProperty.cs

```diff
@@ -10,54 +10,67 @@ namespace Axe.Windows.Core.CustomObjects
 {
     public class CustomProperty
     {
+#pragma warning disable CA1720 // Identifier contains type name: name from JSON
+        private Guid _Guid;
         /// <summary>The RFC4122 globally unique identifier of this property.</summary>
         [JsonProperty("guid")]
-#pragma warning disable CA1720 // Identifier contains type name: name from JSON
-        public Guid Guid { get; set; }
+        public Guid Guid {
+            get { return _Guid; }
+            set
+            {
+                if (value == Guid.Empty) throw new InvalidDataException("Missing GUID in custom property definition.");
```

Comment:
If the file doesn't specify the value, we'll have Guid.Empty, which suggests that we need to check this later on either way. With that in mind, is this validation really useful?